drivers/display/display_led_strip_matrix.c

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2024 TOKITA Hiroshi
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT led_strip_matrix
+
+#include <zephyr/drivers/display.h>
+#include <zephyr/drivers/led_strip.h>
+#include <zephyr/sys/util.h>
+
+#define LOG_LEVEL 4
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(main);
+
+struct led_strip_matrix_config {
+	const struct device *const dev;
+	uint16_t height;
+	uint16_t width;
+	uint16_t module_width;
+	uint16_t module_height;
+	bool circulative;
+	bool start_from_right;
+};
+
+struct led_strip_matrix_data {
+	struct led_rgb *pixels;
+};
+
+static size_t pixel_index(const struct led_strip_matrix_config *config, uint16_t x, uint16_t y)
+{
+	const size_t mod_col = x / config->module_width;
+	const size_t mod_row = y / config->module_height;
+	size_t mod_x = x % config->module_width;
+	const size_t mod_y = y % config->module_height;
+	const size_t mod_pixs = config->module_width * config->module_height;
+	const size_t mod_num_in_row = (config->width / config->module_width);
+	const size_t offset = (mod_num_in_row * mod_row + mod_col) * mod_pixs;
+
+	if (config->circulative) {
+		if (config->start_from_right) {
+			mod_x = (config->module_width - 1) - mod_x;
+		}
+	} else {
+		if ((mod_y % 2) == !config->start_from_right) {
+			mod_x = (config->module_width - 1) - mod_x;
+		}
+	}
+
+	return offset + mod_y * config->module_width + mod_x;
+}
+
+static inline int check_descriptor(const struct led_strip_matrix_config *config, const uint16_t x,
+				   const uint16_t y, const struct display_buffer_descriptor *desc)
+{
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller then width");
+	__ASSERT(desc->pitch <= config->width, "Pitch in descriptor is larger than screen size");
+	__ASSERT(desc->height <= config->height, "Height in descriptor is larger than screen size");
+	__ASSERT(x + desc->pitch <= config->width,
+		 "Writing outside screen boundaries in horizontal direction");
+	__ASSERT(y + desc->height <= config->height,
+		 "Writing outside screen boundaries in vertical direction");
+
+	if (desc->width > desc->pitch || x + desc->pitch > config->width ||
+	    y + desc->height > config->height) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int led_strip_matrix_write(const struct device *dev, const uint16_t x, const uint16_t y,
+				  const struct display_buffer_descriptor *desc, const void *buf)
+{
+	const struct led_strip_matrix_config *config = dev->config;
+	struct led_strip_matrix_data *data = dev->data;
+	const uint32_t *u32buf = buf;
```

Comment:
Yikes, how about calling this `data` or something? Variable names referring to the type looks real bad here. :-)

Reply:
sry not sure why I resolved this one: I think this variable could use a better name, `data`, `pixel_data` or something

Reply:
I didn't touch the 'Resolve conversation' button, so I can't understand why the status was changed.

But I think this issue is solved in the newest version.
I applied your suggestion 
https://github.com/zephyrproject-rtos/zephyr/pull/68614#discussion_r1480174921 and removed the data structure completely. So, the variable was removed.
Currently, variables with the name 'data' do not exist.

I moved the 'data' into this structure, so `struct led_strip_matrix_data` no longer exists.

https://github.com/soburi/zephyr/blob/7d9f1a6256855656ba5b98c2705faab6b557c1bf/drivers/display/display_led_strip_matrix.c#L16-L24

Reply:
oh I was the one to resolve it for some reason :-)

Sure the dev->data pointer is solved but this is specifically asking to rename the u32buf variable. I understand you need a typed variable there but we usually shy away from variable names referencing the type, so I'm suggesting to maybe use "data" for this one instead.

Reply:
I noticed it after I commented! :-)
And I already pushed the fixed version.
We passed each other. Could you check again?

Reply:
Yeah looking good thanks!